src/main/java/chess/domain/Position.java

```diff
@@ -0,0 +1,100 @@
+package chess.domain;
+
+import static java.util.stream.Collectors.toList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.IntStream;
+
+public class Position {
+    private static final Map<Integer, Position> CACHE = new ConcurrentHashMap<>(64);
+    private static final int MIN_RANGE = 0;
+    private static final int MAX_RANGE = 7;
+
+    private final int x;
+    private final int y;
+
+    private Position(int x, int y) {
+        this.x = x;
+        this.y = y;
+    }
+
+    public static Position of(int x, int y) {
+        if (validateRange(x) || validateRange(y)) {
+            throw new IllegalArgumentException("[ERROR] 잘못된 위치 값 입니다.");
+        }
+        return CACHE.computeIfAbsent(Objects.hash(x, y), ignore -> new Position(x, y));
+    }
+
+    private static boolean validateRange(int range) {
+        return range < MIN_RANGE || range > MAX_RANGE;
+    }
+
+    public boolean isSameXTo(Position other) {
```

Comment:
```suggestion
    public boolean isSameXAs(Position other) {
```

영문법상 요게 더 맞는 것 같아요 !